6-AdvancedScenarios/4-hybrid-SPA/App/app.js

```diff
@@ -0,0 +1,51 @@
+require('dotenv').config();
+
+const express = require("express");
+const path = require("path");
+const app = express();
+const expressSession = require('express-session');
+const passport = require('passport');
+const BearerStrategy = require('passport-azure-ad').BearerStrategy;
+const mainRouter = require('./routes/mainRoutes');
+
+app.use(express.json());
+app.use(express.static(path.join(__dirname, 'client/build')));
+app.use(express.urlencoded({ extended: false }));
+
+app.use(expressSession({
+    secret: process.env.CLIENT_SECRET,
+    resave: false,
+    saveUninitialized: false
+}));
+
+const bearerOptions = {
+    identityMetadata: `https://login.microsoftonline.com/${process.env.TENANT_ID}/v2.0/.well-known/openid-configuration`,
+    issuer: `https://login.microsoftonline.com/${process.env.TENANT_ID}/v2.0`,
+    clientID: process.env.CLIENT_ID,
+    audience: process.env.CLIENT_ID, // audience is this application
```

Comment:
This may need to be:

```suggestion
    audience: `api://${process.env.CLIENT_ID}`, // audience is this application
```

Reply:
And considering there can be trailling slashes for the aud value for tokens, I would instead probably have audience passed in explicitly.

Reply:
This format is for access token v1, and it's not applicable for access token v2. We instruct our users to update the access token to v2 in the sample doc.

Reply:
Regardless, our other SDKs have the application pass in valid audiences (multiple) to account for any format and any nuances (e.g. multiple environments, trailing slashes, etc).

Reply:
Ok, I added an environment variable for the audience and an explanation in the readme file.